packages/launch/src/hooks/use-domain-selection.ts

```diff
@@ -1,15 +1,62 @@
 /**
  * External dependencies
  */
+import * as React from 'react';
 import { useDispatch, useSelect } from '@wordpress/data';
 import type { DomainSuggestions } from '@automattic/data-stores';
 
 /**
  * Internal dependencies
  */
-import { LAUNCH_STORE } from '../stores';
+import { LAUNCH_STORE, SITE_STORE, DOMAIN_SUGGESTIONS_STORE } from '../stores';
+import LaunchContext from '../context';
+import { isDomainProduct } from '../utils';
+import type { Product, DomainProduct } from '../utils';
 
-export function useDomainSelection() {
+export function useDomainProductFromCart(): DomainProduct | undefined {
+	const { siteId } = React.useContext( LaunchContext );
+	const { getCart } = useDispatch( SITE_STORE );
+
+	const [ domainProductFromCart, setDomainProductFromCart ] = React.useState<
+		DomainProduct | undefined
+	>( undefined );
+
+	React.useEffect( () => {
+		( async function () {
+			const cart = await getCart( siteId );
+			const domainProduct = cart.products?.find( ( item: Product ) => isDomainProduct( item ) );
+			setDomainProductFromCart( domainProduct );
+		} )();
+	}, [ siteId, getCart, setDomainProductFromCart ] );
+
+	return domainProductFromCart;
+}
+
+export function useDomainSuggestionFromCart(): DomainSuggestions.DomainSuggestion | undefined {
+	const domainProductFromCart = useDomainProductFromCart();
+
+	const domainName = domainProductFromCart?.meta;
+
+	const domainSuggestion = useSelect( ( select ) =>
+		domainName
+			? select( DOMAIN_SUGGESTIONS_STORE ).getDomainSuggestions( domainName, {
+					quantity: 1,
+					include_wordpressdotcom: false,
+					include_dotblogsubdomain: false,
+			  } )?.[ 0 ]
+			: undefined
```

Comment:
This is a ternary operator because `useSelect()` cannot be inside an `if` block.

Reply:
We could use the Logical AND (`&&`) and wrap it in a [void operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void) to get rid of the ternary and have the same effect. 

```ts
void(domainName &&
select( DOMAIN_SUGGESTIONS_STORE ).getDomainSuggestions( domainName, {
   quantity: 1,
   include_wordpressdotcom: false,
   include_dotblogsubdomain: false,
} )?.[ 0 ])
```

Reply:
That's a cool trick. Did a search in Calypso codebase, I don't think many people know about it. Updated to use `void()`.

Reply:
Just an update. The `void()` trick didn't work unfortunately. It always returns `undefined`. Reverting back to original. Sorry mate!